src/uct/tcp/sockcm/sockcm_md.c

```diff
@@ -43,10 +43,105 @@ ucs_status_t uct_sockcm_md_query(uct_md_h md, uct_md_attr_t *md_attr)
     return UCS_OK;
 }
 
+static int uct_sockcm_is_addr_route_resolved(int sock_id, struct sockaddr *addr,
+                                            int addrlen)
+{
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    char host[UCS_SOCKADDR_STRING_LEN];
+    char serv[UCS_SOCKADDR_STRING_LEN];
+    int ret = -1;
+
+    ret = getnameinfo(addr, addrlen, host, UCS_SOCKADDR_STRING_LEN, serv,
+                      UCS_SOCKADDR_STRING_LEN, NI_NAMEREQD);
+    if (0 != ret) {
+        ucs_debug("getnameinfo error : %s\n", gai_strerror(ret));
+        return 0;
+    }
+
+    if (connect(sock_id, addr, addrlen)) {
```

Comment:
If server isn't listening, we get connection refused error (`ECONNREFUSED`) and this function still returns that the server is reachable should that error be detected. That should suffice for reachability right? If the server can't be reached we'll get back `ENETUNREACH` error and we return that the addr isn't reachable by returning 0.
https://github.com/openucx/ucx/pull/3570/files#diff-e6d8a43b8d2e3efb16e0f895a16f4486R63

That said, calling `connect` isn't needed but just calling `getnameinfo` will tell if the given address is reachable from transports that support sockaddr. `getnameinfo` will return an error like:
```
 getnameinfo error : Name or service not known
```
Which should suffice, right? I'll remove the `connect` part. Lmk if this is ok.

Reply:
Since connect() can return even though the server didn't call accept() yet, I agree that connect() isn't a good indication here. I'm not sure though about getnameinfo() - can you please check this with gtest?

Reply:
will do

Reply:
10x! Also, please note, that in order to test sockcm in gtest, will need to make sure that both rdmacm and sockcm are tested everywhere - in uct's gtest, in ucp's gtest, and here as well:
https://github.com/openucx/ucx/blob/master/test/gtest/uct/test_md.cc#L523